src/main/java/io/usethesource/vallang/IConstructor.java

```diff
@@ -25,6 +25,14 @@
  */
 public interface IConstructor extends INode {
 
+	@Override
+	default int getPatternMatchFingerprint() {
+		return getName().hashCode() << 2 + arity();
+
+		// TODO: this would distinguish more constructors based on incomparable argument types:
```

Comment:
I almost committed this, but I realize this has to be reflected in the switch code generator in the compiler as well. So we need to have a java function that produces the hashCode for a given constructor then. We can push this to the future. We only have to realize that changing these hashCodes *breaks all earlier compiled Rascal code, always*.

Reply:
Yes indeed. We already have this at the moment in `util::Reflective` via the functions `getFingerprint` (2 versions) and `getFingerprintNode`. Note that the former has a boolean parameter `concretePatterns`. In the new compiler set-up a single function `getFingerprint(value v)` will do. (In the compiler runtime I have a function `getConcreteFingerprint` that operates on concrete syntax trees.)

Reply:
So leave as is for now and change later?

Reply:
Yes let's do that